spec/gollum_git_repo_modification_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+require 'rspec/expectations'
+
+RSpec::Matchers.define :a_blob_named do |expected|
+  match do |actual|
+    actual.is_a?(Gollum::Git::Blob) && actual.name == expected
+  end
+end
+
+describe Gollum::Git::Repo do
+
+  after(:each) do
+    if parent_commit != repo.head.commit
+      repo.update_ref(repo.head.name, parent_commit.id)
+    end
+  end
+
+  let(:repo) { Gollum::Git::Repo.new(fixture('dot_bare_git'), :is_bare => true) }
+
+  let(:index) do
+    repo.index.read_tree(repo.head.commit.tree.id)
+    repo.index
+  end
+
+  let(:actor) { Gollum::Git::Actor.new("Tom Werner", "tom@example.com") }
+
+  let(:message) { 'Test commit' }
+
+  let(:parent_commit) { repo.head.commit }
+
+  def commit_index
+    index.commit(message, [parent_commit], actor, nil, repo.head.name)
+  end
+
+  describe "#head#commit" do
+
+    context "after committing" do
+
+      before(:each) do
+        index.add('Add.txt', 'Some data')
+        commit_index
```

Comment:
Yes, it does.

At the beginning, I was creating a copy of the fixture repository for each example, and it made the examples slow, so I switched to using `update_ref`.

I guess it would be better to create a copy of the fixture repository before the example group, and use it for all examples in this group, and using `update_ref` to "clean up" between examples?